File: packages/backend/src/main/kotlin/quizzem/backend/features/question/dto/QuestionDto.kt
package quizzem.backend.features.question.dto

import quizzem.backend.features.answer.dto.AnswerDto
import quizzem.backend.features.question.model.QuestionType
import java.time.Instant
import java.util.*

data class QuestionDto(
    val id: UUID,
    val createdAt: Instant,
    val updatedAt: Instant,

    val text: String,
    val type: QuestionType,
    val correctAnswer: String,
    val answers: List<AnswerDto>
)

File: packages/backend2/src/features/question/question.dto.ts
import { QuizzemDto } from '~/src/core/domain/quizzem.dto';
import { AnswerDto } from '~/src/features/answer/answer.dto';
import { QuestionType } from '~/src/features/question/question-type.model';

/**
 * @see QuestionModel
 */
export class QuestionDto extends QuizzemDto {
  text!: string;
  type!: QuestionType;
  correctAnswer!: string;
  answers!: AnswerDto[];
}
